### DISCUSSION TITLE
Any plans to add gamepad support to navigate the interface?

### DISCUSSION THREAD

**Discussion by MathiewMay** [Heroic-Games-Launcher/HeroicGamesLauncher#1111] [Ideas]
Hi, with the steam deck you can install the Heroic Games Launcher with flatpak, when you add it as a non steam game you can start the Heroic Games Launcher from the steam deck interface, it is really seamless, it would be nice if you could control the ui with the gamepad so you would not need to use the touch screen mouse input.

**Comment by flavioislima** [comment]
Heroic already supports navigating with a joystick.
For that to work on the Steam Deck we need the Joystick mappings.
If you provide that to us, we can include in the next release.
Check here on how to provide that to us:
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/wiki/Gamepad-Navigation

**Reply by gazpachoking** [reply]
I tried to get the mappings, gamepad-tester doesn't register anything with the steam deck in desktop mode, or when I load a browser in game mode. Anything else we can do to help get gamepad controls working on steam deck?

**Reply by crazyquark** [reply]
Me too, is there another way? Maybe an app we could run?

**Reply by EMongSTi** [reply]
I am not sure if this is accurate, I had to use Firefox, Chrome seems like it steals the controller input by default. Using Firefox in desktop mode or adding it as a non-steam game and using it in game mode and selecting 'Gamepad With Joystick Trackpad' came up with this. I had to type it out by hand so hopefully I didn't miss anything:

28de-11ff-Microsoft X-Box 360 pad 0
INDEX: 0 CONNECTED: Yes MAPPING: n/a TIMESTAMP: 89039.00000
Pose: n/a HapticActuators: n/a Hand: n/a DisplayId: n/a Vibrations: n/a
B0: 1.00 B1: 0.00 B2: 0.00 B3: 0.00 B4: 0.00 B5: 0.00 B6: 0.00 B7: 0.00 B8: 0.00 B9: 0.00 B10: 0.00
AXIS 0: 0.00000 AXIS 1: 0.00000 AXIS 2: 0.00000 AXIS 3: -0.00101 AXIS 4: 0.06140 AXIS 5: 0.00000 AXIS 6: 0.00000 AXIS 7: 0.00000

**Reply by gazpachoking** [reply]
Did something just change? Maybe a SteamOS update did something. I tried today, and now navigating the interface is working with the buttons. Only works if you have "Web Browser" selected as the controller template though.

**Reply by flavioislima** [reply]
Nice, might be the latest OS update because nothing was changed on our end. :)
